src/sidebar/util/oauth-client.js

```diff
@@ -203,21 +197,22 @@ export default class OAuthClient {
    * Make an `application/x-www-form-urlencoded` POST request.
    *
    * @param {string} url
-   * @param {Object} data - Parameter dictionary
+   * @param {Record<string, string>} data - Parameter dictionary
    */
   _formPost(url, data) {
-    // The `fetch` API has native support for sending form data by setting
-    // the `body` option to a `FormData` instance. We are not using that here
-    // because our test environment has very limited `FormData` support and it
-    // is simpler just to format the data manually.
-    const formData = queryString.stringify(data);
+    const params = new URLSearchParams();
+    for (let [key, value] of Object.entries(data)) {
+      params.set(key, value);
+    }
+    params.sort();
+
     const headers = {
       'Content-Type': 'application/x-www-form-urlencoded',
     };
     return fetch(url, {
       method: 'POST',
       headers,
-      body: formData,
+      body: params.toString(),
```

Comment:
Note that `URLSearchParams.toString` produces output in the same `x-www-form-urlencoded` format as a form post.

Reply:
I wonder if this will trip anyone up in the future the way it gave me pause when I first saw it. Perhaps nothing a quick Google wouldn't sort out.